docs/DeveloperGuide.md

```diff
@@ -273,6 +273,83 @@ the gender is not of valid format; Invalid person exception is thrown if the per
+
+![AddPersonActivityDiagram](images/AddPersonActivityDiagram_DOB.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the
+activities related to date of birth field are considered and shown in this activity diagram.
+</div>
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:**Parser exceptions are thrown and caught if
+date of birth field is not provided in the command, or the date of birth is not of valid format; Duplicated person exception is
+thrown if the person to add already exists in the contact list. Error message is displayed on the GUI subsequently.
+</div>
+
+#### Design considerations:
+
+**Aspect: Whether date of birth field should be optional for a person:**
+
+* **Alternative 1 (current choice):** Compulsory date of birth field:
+    * Pros: It is a more logical implementation because date of birth is a common attribute for all persons,
+  similar to name, address, etc, which are also compulsory field for persons in contact list.
+    * Cons: It is less flexible in cases where the user is missing the date of birth field for a contact,
+    but wants to add the contact anyways, as the user is not able to leave the date of birth field blank.
+
+* **Alternative 2:** Optional date of birth field:
+    * Pros: It is a more flexible implementation, since the user has the choice to set a date of birth,
+    or leave it empty.
+    * Cons: It is less logical since date of birth is a common attribute for a person.
+
+### Edit Date of Birth
+
+The Edit Date of Birth feature allows users to edit a date of birth field (format: dd/mm/yyyy) of a person in the contact list.
+It is performed as a part of `editPersonCommand#execute()`.
+
+These operations are exposed in the `Model` interface as the method `Model#setPerson()`, which calls
+`AddressBook#setPerson()` which calls `UniquePersonList#setPerson()` to replace an existing person with a new person
+object with edited fields in the person list stored in AddressBook.
+
+The following sequence diagram shows the methods calls related to edit person operation:
+
+![EditPersonSequenceDiagram](images/EditPersonSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the edit
+person command text entered by user; the `setPerson(P1, P2)` method replaces person P1 with person P2 in the person list
+in the model. The specific `UniquePersonList` operations are not shown in the diagram for simplicity.
+</div>
+
+The following activity diagram shows what happens when a user executes a new edit command:
+
+![EditPersonActivityDiagram](images/EditPersonActivityDiagram_DOB.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the
+activities related to date of birth field are considered and shown in this activity diagram. All fields are considered optional
```

Comment:
"All fields are considered optional" --> Not entirely true as at least 1 field must be provided and the others may then be optional.